pages/browse/subjects/[...slug].tsx

```diff
@@ -1,19 +1,123 @@
  * filtered by at least one subject heading, as well as displaying and controlling pagination, sort,
  * and other filters.
  */
-export default function SubjectHeadingResults() {
-  const metadataTitle = `Subject Heading Results | ${SITE_NAME}`
+export default function SubjectHeadingResults({
+  bannerNotification,
+  results,
+  isAuthenticated,
+  errorStatus = null,
+  metadataTitle,
+}: SubjectSearchProps) {
+  const { pathname, push, query } = useRouter()
+
+  const { setPersistentFocus } = useFocusContext()
+
+  const slug = Array.isArray(query.slug) ? query.slug[0] : query.slug
+  const subjectFilterKey = "filters[subjectLiteral][0]"
+
+  // Inject SH filter internally if it’s not already in query
+  const initialQuery = {
+    ...query,
+    [subjectFilterKey]: query[subjectFilterKey] ?? slug,
+    page: Array.isArray(query.page) ? query.page[0] : query.page ?? "1",
+  }
+
+  const searchParams = mapQueryToSearchParams(initialQuery)
+
+  const handlePageChange = async (newPage: number) => {
+    setPersistentFocus(idConstants.searchResultsHeading)
+    await push({
+      pathname: pathname,
+      query: {
+        ...query,
+        slug: Array.isArray(query.slug) ? query.slug : [query.slug],
+        page: newPage.toString(),
+      },
+    })
+  }
+
+  const handleSortChange = async (selectedSortOption: string) => {
+    const [sortBy, order] = selectedSortOption.split("_")
+
+    setPersistentFocus(idConstants.searchResultsSort)
+    await push({
+      pathname: pathname,
+      query: {
+        ...query,
+        slug: Array.isArray(query.slug) ? query.slug : [query.slug],
+        sort: sortBy,
+        sort_direction: order,
+        page: "1", // reset page on sort
+      },
+    })
+  }
 
   return (
-    <>
-      <RCHead metadataTitle={metadataTitle} />
-      <Layout activePage="sh-results"></Layout>
-    </>
+    <Search
+      bannerNotification={bannerNotification}
+      results={results}
+      isAuthenticated={isAuthenticated}
+      errorStatus={errorStatus}
+      metadataTitle={metadataTitle}
+      activePage="sh-results"
+      searchParams={searchParams}
+      handlePageChange={handlePageChange}
+      handleSortChange={handleSortChange}
+      getResultsHeading={getBrowseResultsHeading}
+      subjectHeadingSlug={slug}
+    />
   )
 }
+
+export async function getServerSideProps({ req, query, params }) {
+  const bannerNotification = process.env.SEARCH_RESULTS_NOTIFICATION || ""
+  const patronTokenResponse = await initializePatronTokenAuth(req.cookies)
+
+  const baseQuery = buildSubjectQuery({ slug: params.slug, query })
+
+  const results = await fetchResults(mapQueryToSearchParams(baseQuery))
+
+  if (results.status !== 200) {
+    return { props: { errorStatus: results.status } }
+  }
+
+  const redirect = checkForRedirectOnMatch(results, query)
+  if (redirect) return { redirect }
+
+  return {
+    props: {
```

Comment:
if you coerce `slug` into an array and pass as a prop would that allow you to remove the multiple `isArray` checks on the component and in `buildSubjectQuery`?

Reply:
I also think slug might not be necessary in the query params. You should be able to rely on its presence in the URL. You can include it in the `pathname` that you are pushing on page and sort changes. it's hard to tell with the ES index acting so erratically, but I'm pretty sure it doesn't need to be in both places (query and url param)